### PR TITLE
Update OpenShift doc to account for new environment variable options

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 1]
```diff
@@ -193,39 +193,152 @@ To add an annotation in the generated resources:
...
+===== Validation
+
+A conflict between two definitions, e.g. mistakenly assigning both a value and specifying that a variable is derived from a field, will result in an error being thrown at build time so that you get the opportunity to fix the issue before you deploy your application to your cluster where it might be more difficult to diagnose the source of the issue.
+
+Similarly, two redundant definitions, e.g. defining an injection from the same secret twice, will not cause an issue but will indeed report a warning to let you know that you might not have intended to duplicate that definition.
+
+[#env-vars-backwards]
+===== Backwards compatibility
+
+Previous versions of the OpenShift extension supported a different syntax to add environment variables.The older syntax is still supported but is deprecated and it's advised that you migrate to the new syntax.
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:325]:
```suggestion
Previous versions of the OpenShift extension supported a different syntax to add environment variables. The older syntax is still supported but is deprecated and it's advised that you migrate to the new syntax.
```

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 2]
```diff
@@ -193,39 +193,152 @@ To add an annotation in the generated resources:
...
+| All from `ConfigMap`          |`quarkus.openshift.env-vars.xxx.configmap=foobar`     | `quarkus.openshift.env.configmaps=foobar`          |
+| All from `Secret`             |`quarkus.openshift.env-vars.xxx.secret=foobar`        | `quarkus.openshift.env.secrets=foobar`             |
+| From one `Secret` field       |`quarkus.openshift.env-vars.foo.secret=foobar`        | `quarkus.openshift.env.mapping.foo.from-secret=foobar` |
+|                               |`quarkus.openshift.env-vars.foo.value=field`          | `quarkus.openshift.env.mapping.foo.with-key=field` |
+| From one `ConfigMap` field    |`quarkus.openshift.env-vars.foo.configmap=foobar`     | `quarkus.openshift.env.mapping.foo.from-configmap=foobar` |
+|                               |`quarkus.openshift.env-vars.foo.value=field`          | `quarkus.openshift.env.mapping.foo.with-key=field` |
+|====
+
+NOTE: If you redefine the same variable using the new syntax while keeping the old syntax, **ONLY** the new version will be kept and a warning will be issued to alert you of the problem.For example, if you define both
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:340]:
```suggestion
NOTE: If you redefine the same variable using the new syntax while keeping the old syntax, **ONLY** the new version will be kept and a warning will be issued to alert you of the problem. For example, if you define both
```

**Review by gsmet — DISMISSED** [review]
Apparently, you don't like spaces after a dot :).

**Comment by metacosm** [comment]
That is weird…

**Comment by geoand** [comment]
Some IDE setting maybe?

**Comment by metacosm** [comment]
> Some IDE setting maybe?

More like a bug in the IDEA asciidoc plugin because it's not really consistent… Not sure what's going on there but that is weird.

**Comment by geoand** [comment]
> Another weird thing is that it doesn't even show me that one of these lines was modified in the last commit so there's something funky going on…

Maybe you have configured it to not show whitespace changes?

**Comment by metacosm** [comment]
> > Another weird thing is that it doesn't even show me that one of these lines was modified in the last commit so there's something funky going on…
> 
> Maybe you have configured it to not show whitespace changes?

No, it's not that, it won't even show the line was changed when using `Annotate with git blame`.

**Review by gsmet — APPROVED** [review]

**Pull request merged**